### PR TITLE
Clippy + Typos

### PR DESCRIPTION
**Pull request by ModProg**
- **Fix clippy**
- **Fix typos**

### CODE REVIEW

File: src/data.rs [hunk 1]
```diff
@@ -69,7 +70,7 @@ pub enum SimpleType<'a> {
 	/// Tuple struct or tuple variant.
 	Tuple(&'a Fields<'a>),
 	/// Union.
-	Union(&'a Fields<'a>),
+	Union(#[allow(unused)] &'a Fields<'a>),
```
Comment (daxpedda) [inline src/data.rs:73]: If its unused, can we remove it?
Or whats going on here?

(sorry, didn't check myself yet)

Comment (ModProg) [reply src/data.rs:73]: yeah, not sure why we even have it in this enum, probably because we are parsing it?

Comment (daxpedda) [reply src/data.rs]: Yep, removed!

**Comment by daxpedda** [comment]
So before you laugh at me: yes, I do actually think there is something as too much `cfg` guards.
I didn't think before pushing this though.

Let me know if you die in apprehension or actually like it.

**Review by daxpedda — APPROVED** [review]

**Pull request merged**